client/src/components/PopoverOptionSet/tests/PopoverOptionSet-test.js

```diff
@@ -1,148 +1,114 @@
+  render(
+    <PopoverOptionSet {...makeProps({
+      toggle
+    })}
+    />
+  );
+  const popover = await screen.findByTestId('test-popover');
+  fireEvent.click(popover);
+  expect(toggle).toHaveBeenCalled();
+});
 
-      wrapper.setState({ searchValue: 'something ' });
-      wrapper.instance().handleSearchValueClear();
-      expect(wrapper.state('searchValue')).toEqual('');
-    });
-  });
+test('PopoverOptionSet handleSearchValueClear should set the state', async () => {
+  render(
+    <PopoverOptionSet {...makeProps()}/>
+  );
+  const popover = await screen.findByTestId('test-popover');
+  const input = popover.querySelector('input.popover-option-set__search-input');
+  expect(screen.queryByText('No results found')).toBeNull();
+  fireEvent.change(input, { target: { value: 'something' } });
+  const results = await screen.findByText('No results found');
+  expect(results).not.toBeNull();
+});
 
-  describe('handleSearchValueChange', () => {
-    it('should update the internal state on user input change', () => {
-      const wrapper = shallow(
-        <PopoverOptionSet {...props} />
-      );
+test('PopoverOptionSet handleSearchValueClear should set the state', async () => {
+  render(
+    <PopoverOptionSet {...makeProps()}/>
+  );
+  const popover = await screen.findByTestId('test-popover');
+  const input = popover.querySelector('input.popover-option-set__search-input');
+  expect(screen.queryByText('Clear')).toBeNull();
+  fireEvent.change(input, { target: { value: 'something' } });
+  const button = await screen.findByText('Clear');
+  expect(button.tagName).toBe('BUTTON');
+});
 
-      const event = {
-        target: {
-          value: 'a'
+test('PopoverOptionSet renderOptionButtons render all available buttons', async () => {
+  render(
+    <PopoverOptionSet {...makeProps({
+      onSearch: () => [
+        {
+          key: 'a',
+          content: 'A',
+          className: 'dummy-classname-a',
+          onClick: () => null,
+        },
+        {
+          key: 'b',
+          content: 'B',
+          className: 'dummy-classname-b',
+          onClick: () => null,
         }
-      };
-
-      wrapper.instance().handleSearchValueChange(event);
-      expect(wrapper.state('searchValue')).toEqual('a');
-    });
-  });
-
-  describe('renderSearchValueClearLink', () => {
-    it('should render a link to clear the search field', () => {
-      const wrapper = shallow(
-        <PopoverOptionSet {...props} />
-      );
-      wrapper.setState({ searchValue: 'something' });
-      expect(wrapper.find('.popover-option-set__search-clear').length).toEqual(1);
-    });
-  });
-
-  describe('renderOptionButtons', () => {
-    it('render all available buttons', () => {
-      const wrapper = shallow(
-        <PopoverOptionSet {...props} />
-      );
-      wrapper.setState({ searchValue: '' });
-      expect(wrapper.find(Button).length).toEqual(2);
-    });
-
-    it('render all buttons matching the search term', () => {
```

Comment:
These two tests would rely on logic passed in from the onSearch callback i.e. these tests wouldn't be testing anything within the actual code